test/e2e/basic/csvs.go

```diff
@@ -0,0 +1,228 @@
+		BeforeEach(func() {
+			var err error
+
+			By("Getting new client")
+			cli, err = newClient()
+			Expect(err).To(Succeed())
+
+			olmoperators.Install(cli.Scheme())
+
+			By("Listing all current CSVs")
+			performanceOperatorCSVs, err = getCurrentCSV(cli, context.TODO())
+			Expect(err).To(Succeed())
+
+			existingCSVs = len(performanceOperatorCSVs)
+
+			By(fmt.Sprintf("Delete all(%d) CSVs", existingCSVs))
+			for _, element := range performanceOperatorCSVs {
+				err = cli.Delete(context.TODO(), &element)
+				Expect(err).To(Succeed())
+			}
+
+			paginationLimit = uint64(1000)
+			By(fmt.Sprintf("Setting default paginationLimit to: %d", paginationLimit))
+		})
+		AfterEach(func() {
+			var err error
+
+			By(fmt.Sprintf("Recreate all(%d) previously existing CSVs", len(performanceOperatorCSVs)))
+			created := 0
+			for _, element := range performanceOperatorCSVs {
+				err = cli.Create(context.TODO(), &element)
+				Expect(err).To(Succeed())
+				created++
+			}
+			Expect(created).To(Equal(existingCSVs))
+		})
+
+		When("there is no CSVs in the cluster", func() {
+			It("Should list CSVs without timeout", func() {
+				options := []client.ListOption{}
+				ret, err := paocontroller.ListPerformanceOperatorCSVs(cli, options, paginationLimit)
+				Expect(err).To(Succeed())
+				Expect(ret).To(HaveLen(0))
+
+			})
+		})
+
+		When("there are few CSVs in the cluster", func() {
+			const csvs_file = "../testing_manifests/csv-dummy.yaml"
+			const numberCSVsToCreate = 100
+			var csvs_created []types.NamespacedName
+			BeforeEach(func() {
+				var err error
+
+				By("reading file")
+				b, err := os.ReadFile(csvs_file)
+				Expect(err).To(Succeed())
+
+				By("unmarshaling file")
+				csvsOrig := olmv1alpha1.ClusterServiceVersion{}
+				err = yaml.Unmarshal(b, &csvsOrig)
+				Expect(err).To(Succeed())
+
+				By(fmt.Sprintf("Create some(%d) CSVs", numberCSVsToCreate))
+				for idx := 0; idx < numberCSVsToCreate; idx++ {
+					csvs := csvsOrig.DeepCopy()
+					csvs.Name = fmt.Sprintf("csvs-test-%d", idx)
+					csvs.Namespace = ntoconfig.WatchNamespace()
+
+					err = cli.Create(context.TODO(), csvs)
+					Expect(err).To(Succeed())
+					key := types.NamespacedName{
+						Name:      csvs.Name,
+						Namespace: csvs.Namespace,
+					}
+					csvs_created = append(csvs_created, key)
+				}
+				Expect(len(csvs_created)).To(Equal(numberCSVsToCreate))
+			})
+
+			AfterEach(func() {
+				var err error
+
+				By(fmt.Sprintf("Delete all(%d) previously created CSVs", len(csvs_created)))
+				for _, key := range csvs_created {
+					obj := olmv1alpha1.ClusterServiceVersion{}
+					err = cli.Get(context.TODO(), key, &obj)
+					Expect(err).To(Succeed())
+
+					err = cli.Delete(context.TODO(), &obj)
+					Expect(err).To(Succeed())
+				}
+				csvs_created = nil
+			})
+
+			When("pagination value is set to ONE", func() {
+				var oldPaginationLimit uint64
+				BeforeEach(func() {
+					oldPaginationLimit = paginationLimit
+					paginationLimit = 1
+				})
```

Comment:
can we just shadow the pagination limit here in the innermost closure? would it be less readable? (I don't think so, but let's see)